### ISSUE TITLE
Too many possible values for `class` and loss of POI category due to promotion of `subclass` value

### ISSUE THREAD

**Issue by ghost** [openmaptiles/openmaptiles#1557]
https://openmaptiles.org/schema/#fields says that "class" has a fixed set of possible values, however, if there's no `class`, then it will be the value of `subclass`.

However `subclass` has the original OSM value of fields like `leisure` which itself has 113 options: https://taginfo.openstreetmap.org/keys/leisure#values

Example; the feature for https://www.openstreetmap.org/node/421419270 is:

```
  "class": "hackerspace",
  "subclass": "hackerspace",
```

As many map styles will likely choose the POI image from the `class`, this adds a considerable number of icons a map designer would have to provide (200+ icons).

Currently, the information that "hackerspace" is of type "leisure" is lost, so the map renderer would have to map back these names to the type (which might not even be possible due to ambiguity in the field values which can be promoted to `class`/`subclass`)

I think this is a bug in the spec, because `class` should probably just be `leisure` (the source of the `subclass` value) then, i.e.:

```
  "class": "leisure",
  "subclass": "hackerspace",
```

---

This would mean the only possible values for `class` would be:

**Originally specified:**
1. shop
1. town_hall
1. golf
1. fast_food
1. park
1. bus
1. railway
1. aerialway
1. entrance
1. campsite
1. laundry
1. grocery
1. library
1. college
1. lodging
1. ice_cream
1. post
1. cafe
1. school
1. alcohol_shop
1. bar
1. harbor
1. car
1. hospital
1. cemetery
1. attraction
1. beer
1. music
1. stadium
1. art_gallery
1. clothing_store
1. swimming
1. castle
1. atm

**Promoted from *source* of `subclass` (some existed in `class` already):**
1. amenity
1. barrier
1. historic
1. information
1. landuse
1. leisure
1. ~~railway~~
1. ~~shop~~
1. sport
1. station
1. religion
1. tourism
1. ~~aerialway~~
1. building
1. highway
1. office
1. waterway

**Comment by ZeLonewolf** [comment]
> Instead of having a couple of classes, OMT currently has 200+ different "class" values.. and that defeats the purpose of class/subclass. I also don't think it's a good idea to remap all 113 values (for "leisure" alone, let alone the other keys being promoted to subclass) to new classes as part of OMT. - it's not just "some of the POIs" (as you said on Slack), it's literally hundreds, if not thousands!

I disagree, I think that's exactly what OMT should be doing. `class=leisure` isn't useful to a style author - it's not likely that you would want to style all `lesiure=*` with common styling. The whole point of the `class` hierarchy is to group together elements that might share common styling and `leisure`, `amenity`, etc., are too diverse to use them as-is. Now maybe 200 is too many and it should be 100 or 50 or whatever, but thematic grouping is absolutely a desired feature for style authoring.
